pkg/controller/storagecluster/external_resources.go

```diff
@@ -161,16 +196,18 @@ func (r *ReconcileStorageCluster) newExternalCephObjectStoreInstances(
 // ensureExternalStorageClusterResources ensures that requested resources for the external cluster
 // being created
 func (r *ReconcileStorageCluster) ensureExternalStorageClusterResources(instance *ocsv1.StorageCluster, reqLogger logr.Logger) error {
-	// check for the status boolean value accepted or not
-	if instance.Status.ExternalSecretFound {
+	if r.sameExternalSecretData(instance) {
 		return nil
 	}
 	err := r.createExternalStorageClusterResources(instance, reqLogger)
 	if err != nil {
 		reqLogger.Error(err, "could not create ExternalStorageClusterResource")
 		return err
 	}
-	instance.Status.ExternalSecretFound = true
+	// as there are no errors, update the external checksum status
+	if externalSecretDataHash, err := r.externalSecretDataChecksum(instance); err == nil {
```

Comment:
Should something happen if `err != nil`? I'd think we can't rightly say we succeeded if we can't calculate the new checksum...

Reply:
The original idea was, as we have updated all the resources returning an error here doesn't make any sense. In worse case scenario we will be reconciling again and again, which we will do any way (even after sending an error, as we are unable to set the newer checksum into the status).